tests/unit/test.config.js

```diff
@@ -296,6 +296,30 @@ describe('config', () => {
       assert.strictEqual(argv.ignoreFiles, configObject.ignoreFiles);
     });
 
+    it('does not mistake an array config values for a sub-command',
+       () => {
+         const params = makeArgv({
+           userCmd: ['fakecommand'],
+           globalOpt: {
+             pref: {
+               demand: false,
+               type: 'string',
```

Comment:
Shouldn't this be `type: 'array'` so that it tests the bug you were fixing more directly? I'm a bit confused because the exception message in your assertion doesn't seem to match the spec of this test.